examples/pipeline_wavernn/datasets.py

```diff
@@ -0,0 +1,115 @@
+import os
+import random
+
+import torch
+import torchaudio
+from torch.utils.data.dataset import random_split
+from torchaudio.datasets import LJSPEECH
+from torchaudio.transforms import MuLawEncoding
+
+from processing import encode_waveform_into_bits, encode_bits_into_waveform
+
+
+class MapMemoryCache(torch.utils.data.Dataset):
+    r"""Wrap a dataset so that, whenever a new item is returned, it is saved to memory.
+    """
+
+    def __init__(self, dataset):
+        self.dataset = dataset
+        self._cache = [None] * len(dataset)
+
+    def __getitem__(self, n):
+        if self._cache[n] is not None:
+            return self._cache[n]
+
+        item = self.dataset[n]
+        self._cache[n] = item
+
+        return item
+
+    def __len__(self):
+        return len(self.dataset)
+
+
+class Processed(torch.utils.data.Dataset):
+    def __init__(self, dataset, transforms):
+        self.dataset = dataset
+        self.transforms = transforms
+
+    def __getitem__(self, key):
+        item = self.dataset[key]
+        return self.process_datapoint(item)
+
+    def __len__(self):
+        return len(self.dataset)
+
+    def process_datapoint(self, waveform):
```

Comment:
nit: change variable name `waveform` to `item`

Reply:
Yes, it has been changed.